doc/api/addons.markdown

```diff
@@ -157,21 +185,28 @@ function calls and return a result. This is the main and only needed source
     using v8::String;
     using v8::Value;
 
+    // This is the implementation of the "add" method
+    // Input arguments are passed using the
+    // const FunctionCallbackInfo<Value>& args struct
     void Add(const FunctionCallbackInfo<Value>& args) {
       Isolate* isolate = args.GetIsolate();
 
+      // Check the number of arguments passed.
       if (args.Length() < 2) {
+        // Throw an Error that is passed back to JavaScript
         isolate->ThrowException(Exception::TypeError(
             String::NewFromUtf8(isolate, "Wrong number of arguments")));
         return;
       }
 
+      // Check the argument types
       if (!args[0]->IsNumber() || !args[1]->IsNumber()) {
         isolate->ThrowException(Exception::TypeError(
             String::NewFromUtf8(isolate, "Wrong arguments")));
         return;
       }
 
+      // Perform the operation and set the return value
```

Comment:
How about removing "and set the return value" and move it down to where it's set cause it's worth pointing out that "the const FunctionCallbackInfo<Value>& args struct" is _also_ used to set return values, not just pass arguments.